qchem/pennylane_qchem/qchem/obs.py

```diff
@@ -444,10 +444,316 @@ def particle_number(orbitals, mapping="jordan_wigner", wires=None):
+
+    Args:
+        matrix_elements (array[float]): 2D NumPy array with the matrix elements
+            :math:`\langle \alpha \vert \hat{t} \vert \beta \rangle`
+        core (list): indices of core orbitals, i.e., the orbitals that are
+            not correlated in the many-body wave function
+        active (list): indices of active orbitals, i.e., the orbitals used to
+            build the correlated many-body wave function
+        cutoff (float): Cutoff value for including matrix elements. The
+            matrix elements with absolute value less than ``cutoff`` are neglected.
+
+    Returns:
+        tuple: Table of indices and matrix elements of the one-particle operator
+        and the contribution due to core orbitals. The returned table is a 2D Numpy
+        array where each row contains three elements, the *spin*-orbital indices
+        :math:`\alpha, \beta` and the matrix element
+        :math:`\langle \alpha \vert \hat{t} \vert \beta \rangle`.
+
+    **Example**
+
+    >>> matrix_elements = np.array([[-4.72739313e+00, -1.05499666e-01, -1.66961416e-01,
+    ...                               6.18014041e-16,  2.86964662e-16, -3.46772026e-02],
+    ...                             [-1.05499666e-01, -1.49264622e+00, 3.28928073e-02,
+    ...                              -2.20398308e-16,  1.93277291e-16, 5.27078882e-02],
+    ...                             [-1.66961416e-01,  3.28928073e-02, -1.12554473e+00,
+    ...                              -2.82912389e-17,  2.55224784e-16, -3.04455743e-02],
+    ...                             [ 6.18014041e-16, -2.20398308e-16, -2.82912389e-17,
+    ...                              -1.13579985e+00, -1.94289029e-16, -2.36158697e-16],
+    ...                             [ 2.86964662e-16,  1.93277291e-16,  2.55224784e-16,
+    ...                              -2.77555756e-16, -1.13579985e+00,  2.06665432e-16],
+    ...                             [-3.46772026e-02,  5.27078882e-02, -3.04455743e-02,
+    ...                              -2.36158697e-16,  2.06665432e-16, -9.50966595e-01]]
+
+    >>> table, t_core = one_particle(matrix_elements, core=[0], active=[1, 2])
+    >>> print(table)
+    [[ 0.          0.         -1.49264622]
+     [ 1.          1.         -1.49264622]
+     [ 0.          2.          0.03289281]
+     [ 1.          3.          0.03289281]
+     [ 2.          0.          0.03289281]
+     [ 3.          1.          0.03289281]
+     [ 2.          2.         -1.12554473]
+     [ 3.          3.         -1.12554473]]
+    >>> print(t_core)
+    -9.45478626
+    """
+
+    orbitals = matrix_elements.shape[0]
+
+    if matrix_elements.ndim != 2:
+        raise ValueError(
+            "'matrix_elements' must be a 2D array; got matrix_elements.ndim = {}".format(
+                matrix_elements.ndim
+            )
+        )
+
+    if core is None:
+        t_core = 0
+    else:
+        if True in [i > orbitals - 1 or i < 0 for i in core]:
```

Comment:
Minor thing: could it be `any([i > orbitals - 1 or i < 0 for i in core])` which might be easier to read